scripts/dev/dockerutil.py

```diff
@@ -28,6 +28,42 @@ def push_image(tag: str) -> None:
         print("\r" + push_image_formatted(line), end="", flush=True)
 
 
+def retag_image(
+    old_repo_url: str,
+    new_repo_url: str,
+    old_tag: str,
+    new_tag: str,
+    path: str,
+    labels: Optional[dict] = None,
+    username: Optional[str] = None,
+    password: Optional[str] = None,
+    registry: Optional[str] = None,
+) -> None:
+    with open(f"{path}/Dockerfile", "w") as f:
+        f.write(f"FROM {old_repo_url}:{old_tag}")
+    client = docker.from_env()
+    if username is not None and password is not None and registry is not None:
```

Comment:
[nit] this sort of multiple None check where each condition is the same can be written in a way that is a little more pythonic if you prefer it!

```python
if all(value is not None for value in [username, password, registry]):
   # login
```